src/mempool_args.h

```diff
@@ -5,18 +5,23 @@
 #ifndef BITCOIN_MEMPOOL_ARGS_H
 #define BITCOIN_MEMPOOL_ARGS_H
 
+#include <optional>
+
 class ArgsManager;
+class CChainParams;
+struct bilingual_str;
 namespace kernel {
 struct MemPoolOptions;
 };
 
 /**
  * Overlay the options set in \p argsman on top of corresponding members in \p mempool_opts.
+ * Returns an error if one was encountered.
  *
  * @param[in]  argsman The ArgsManager in which to check set options.
  * @param[in,out] mempool_opts The MemPoolOptions to modify according to \p argsman.
  */
-void ApplyArgsManOptions(const ArgsManager& argsman, kernel::MemPoolOptions& mempool_opts);
+[[nodiscard]] std::optional<bilingual_str> ApplyArgsManOptions(const ArgsManager& argsman, const CChainParams& chainparams, kernel::MemPoolOptions& mempool_opts);
```

Comment:
In commit "Return optional error from ApplyArgsManOptions" (fa468bdfb62dec286cb977db78d3e47b64dafeba)

This seems good for now, but after #25665 this could return `util::Result<void>` instead of `std::optional<bilingual_str>`

Reply:
Yes, there are also a few other places. See `git grep 'std::optional<bilingual_str>'`.